node/config.go

```diff
@@ -70,6 +72,12 @@ func WithEnableDevelopment(enable bool) NodeOpt {
 	}
 }
 
+// WithNodeIdentity sets the identity for the node. This is the identity that
+// will be used for things like block signatures and P2P sync operations.
+func WithNodeIdentity(ident identity.Identity) db.Option {
```

Comment:
suggestion: If it is simple to do, can you please use this in the test framework instead of the internal call you mentioned in the PR description?